jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -192,11 +195,24 @@ public void close() {
     @Override
     public io.opentracing.Tracer.SpanBuilder addReference(
         String referenceType, io.opentracing.SpanContext referencedContext) {
-      if (referencedContext instanceof SpanContext
-          && (Utils.equals(referenceType, References.CHILD_OF)
-                      || Utils.equals(referenceType, References .FOLLOWS_FROM))) {
-        this.references.add(new Reference((SpanContext) referencedContext, referenceType));
-        this.baggage.putAll(((SpanContext)referencedContext).baggage());
+
+      if (!(referencedContext instanceof SpanContext)) {
+        return this;
+      }
+
+      // TODO UI currently does not support arbitrary references
```

Comment:
Well, this was based on the previous comment that UI does not support user defined reference types. Will remove.